### DISCUSSION TITLE
How to call a method at a specific time between scheduling intervals?

### DISCUSSION THREAD

**Discussion by EhsanRasoulpour** [manoelcampos/cloudsim-plus#363] [Q&A] [answered]
Hello
for example if we consider scheduling interval equal to 300 s and a delay equal to 100 s between when we set host's state to off and the time it really turns off, the delay is less than a scheduling interval. either i use a condition or i create an Event Listener for changing the state of the host after elapsing the delay, the state will change at the start of next time slot. is there a way to do it right after 100 s?

**Comment by manoelcampos** [comment, marked as answer]
Hello @EhsanRasoulpour 
There is no way to ensure that. If it was the other way around (100s scheduling interval and 300s delay), it would work.
You can use the Simulation onClockTickListener to track the clock, but without guarantees the clock will tick exactly at the time you want. That is why you usually should check if the clock is inside some time interval to fire the desired event/run your code.

Your clock tick listener could be something like this:

```java
private void onClockTickEvent(EventInfo info){
    if(info.getTime() - this.lastDesiredEventTime >= DESIRED_TIME_INTERVAL) {
        //run the desired code 

       //store the time the desired event was fired and your code was ran.
       this.lastDesiredEventTime = info.getTime();
    }
}
```

I hope it helps. In this case, please don't forget to mark the question as answered and give CloudSim Plus a star.
I appreciate it.

**Reply by EhsanRasoulpour** [reply]
thank you for your answer. i used this method and changed my scheduling interval to the value of the least delay time in order to change the state of the host. i think this would also increase the simulation time.

**Reply by manoelcampos** [reply]
Great. Please consider giving the project a star to help promoting it.
